controllers/apps/components/replication/replication_utils.go

```diff
@@ -204,55 +204,44 @@ func generateReplicationSerialPlan(plan *util.Plan, pods []corev1.Pod, rolePrior
 // unknown & empty & 1/2 secondaries -> 1/2 secondaries -> primary
 func generateReplicationBestEffortParallelPlan(plan *util.Plan, pods []corev1.Pod, rolePriorityMap map[string]int) {
 	start := plan.Start
-	// append unknown, empty
-	index := 0
+	unknownEmptySteps := make([]*util.Step, 0)
+	secondarySteps := make([]*util.Step, 0)
+	primarySteps := make([]*util.Step, 0)
```

Comment:
due to collection might be small and pointer type, I would write following version to allocate extra spaces to avoid slice memory reallocation operation:
```go
l := len(pods)
unknownEmptySteps := make([]*util.Step, 0, l)
secondarySteps := make([]*util.Step, 0, l)
primarySteps := make([]*util.Step, 0, l)
```